app/models/abuse_report.rb

```diff
@@ -6,24 +6,29 @@ class AbuseReport < ActiveRecord::Base
   validates_presence_of :summary
   validates_presence_of :comment
   validates_presence_of :url
-  validate :work_is_not_over_reported
+  validate :url_is_not_over_reported
   validates_length_of :summary, maximum: ArchiveConfig.FEEDBACK_SUMMARY_MAX,
                                 too_long: ts('must be less than %{max}
                                              characters long.',
                                 max: ArchiveConfig.FEEDBACK_SUMMARY_MAX_DISPLAYED)
 
   scope :by_date, order('created_at DESC')
 
-  # if the URL ends like "works/123", add a / at the end
-  # if the URL contains "works/123?", remove the parameters and add a /
-  # work_is_not_over_reported uses the / so "/works/1234" isn't a match
+  # Clean work or profile URLs so we can prevent the same URLs from
+  # getting reported too many times.
+  # If the URL ends without a / at the end, add it:
+  # url_is_not_over_reported uses the / so "/works/1234" isn't a match
   # for "/works/123"
-  before_validation :clean_work_url, on: :create
-  def clean_work_url
-    if url.match(/(works\/\d+)$/)
-      self.url = url + '/'
-    elsif url.match(/(works\/\d+\?)/)
-      self.url = url.split('?').first + '/'
+  before_validation :clean_url, on: :create
+  def clean_url
+    # Work URLs: "works/123"
+    # Profile URLs: "users/username"
+    if url.match(/(works\/\d+)/) || url.match(/(users\/\w+)/)
+      uri = Addressable::URI.parse url
+      uri.query = nil
+      uri.fragment = nil
+      uri.path += "/" unless uri.path.end_with? "/"
+      self.url = uri.to_s
```

Comment:
Users cannot submit reports for ao3.org links because of `validates_format_of :url, with: app_url_regex` which uses `ArchiveConfig.APP_HOST`. The `url_is_not_over_reported` validation only looks at the "/works/123" part, so we don't need to clean up the link's protocol.